src/AlgebraicGeometry/Schemes/AffineSchemes/Objects/Methods.jl

```diff
@@ -199,6 +199,67 @@ function components(X::AbsSpec)
   return result
 end
 
+########################################################################
+# Base change and reduction modulo p
+########################################################################
+
+@doc raw"""
+    change_base(phi::Any, X::AbsSpec)
+
+For an affine scheme `X` over a `base_ring` ``𝕜`` and a morphism 
```

Comment:
```suggestion
For an affine scheme `X` over a `base_ring` ``𝕜`` and a morphism 
```
The notation `k` and `K` suggests a field. Is `B` or `R` an option?

Reply:
We have the name `change_base_ring` already and apparently also the function `base_ring` so that name seems appropriate to me? Less names to remember. And it would be consistent with our current use.

Reply:
A less rigorous but common situation is when we have a scheme over `QQ` and want to reduce that mod `p`  and get an error if we divide by zero.  This depends on the chosen generators for the ideal involved of course.
Looks to me like this code can be used for this situation as well?

Reply:
There is a discussion on this on Slack. I'd be open to make these methods of the already existing function `change_base_ring`, but it's not yet completely clear whether we want that. 

@thofma : Any suggestions?

Reply:
> A less rigorous but common situation is when we have a scheme over `QQ` and want to reduce that mod `p` and get an error if we divide by zero. This depends on the chosen generators for the ideal involved of course.
> Looks to me like this code can be used for this situation as well?

That is a good example for a case where you really can not have `phi` as a valid `Hecke.Map` (with domain and codomain as discussed below). Yes, this code should be able to accomodate that. You just have to pass some function that does the job for you to try to reduce mod p and throw an error if something goes wrong.

Reply:
O.k. It seems that `base_change` exists already for elliptic and hyperelliptic curve. It is also used in the literature. So we can stick with that, especially if `change_base_ring` is a little controversial. 
So please go ahead with this ticket.

Reply:
No suggestions.

I think `base_change` is also more appropriate if we have relative schemes and want to change the base for those?

Reply:
But should it then be `base_change` or `change_base`?

Reply:
Looks like `saturation` versus `saturate`. We are not globally consistent here. Is there any guideline?
I would go for `base_change`, also because that already exists. But I do not have a strong opinion.